src/Http/Http.Abstractions/src/HttpResponse.cs

```diff
@@ -16,14 +15,20 @@ namespace Microsoft.AspNetCore.Http
     public abstract class HttpResponse
     {
         private static readonly Func<object, Task> _callbackDelegate = callback => ((Func<Task>)callback)();
-        private static readonly Func<object, Task> _disposeDelegate = disposable =>
+        private static readonly Func<object, Task> _disposeDelegate = state =>
```

Comment:
```suggestion
        private static readonly Func<object, Task> _disposeDelegate = static state =>
```
Is C# 9 usable?

BTW: can the _disposeDelegate return a ValueTask? Or are there multiple awaiters, etc. in play so this must be a Task?

Reply:
The API is Task only

Reply:
Ah, the `OnCompleted`-API...thanks!